src/test/clip.t.sol

```diff
@@ -473,6 +438,109 @@ contract DutchClipperTest is DSTest {
         assertEq(art, 0 ether);
     }
 
+    function test_Hole_hole() public {
+        assertEq(dog.Dirt(), 0);
+        (,,, uint256 dirt) = dog.ilks(ilk);
+        assertEq(dirt, 0);
+
+        dog.bark(ilk, me);
+
+        (, uint256 tab,,,,) = clip.sales(1);
+
+        assertEq(dog.Dirt(), tab);
+        (,,, dirt) = dog.ilks(ilk);
+        assertEq(dirt, tab);
+
+        bytes32 ilk2 = "silver";
+        Clipper clip2 = new Clipper(address(vat), address(spot), address(dog), ilk2);
+        clip2.rely(address(dog));
+
+        dog.file(ilk2, "clip", address(clip2));
+        dog.file(ilk2, "chop", 1.1 ether);
+        dog.file(ilk2, "hole", rad(1000 ether));
+        dog.rely(address(clip2));
+
+        vat.init(ilk2);
+        vat.rely(address(clip2));
+        vat.file(ilk2, "line", rad(100 ether));
+
+        vat.slip(ilk2, me, 40 ether);
+
+        DSValue pip2 = new DSValue();
+        pip2.poke(bytes32(uint256(5 ether))); // Spot = $2.5
+
+        spot.file(ilk2, "pip", address(pip2));
+        spot.file(ilk2, "mat", ray(2 ether));
+        spot.poke(ilk2);
+        vat.frob(ilk2, me, me, me, 40 ether, 100 ether);
+        pip2.poke(bytes32(uint256(4 ether))); // Spot = $2
+        spot.poke(ilk2);
+
+        dog.bark(ilk2, me);
+
+        (, uint256 tab2,,,,) = clip2.sales(1);
+
+        assertEq(dog.Dirt(), tab + tab2);
+        (,,, dirt) = dog.ilks(ilk);
+        (,,, uint256 dirt2) = dog.ilks(ilk2);
+        assertEq(dirt, tab);
+        assertEq(dirt2, tab2);
+    }
+
+    function test_partial_liquidation_Hole_limit() public {
+        dog.file("Hole", rad(75 ether));
+
+        assertEq(_ink(ilk, me), 40 ether);
+        assertEq(_art(ilk, me), 100 ether);
+
+        assertEq(dog.Dirt(), 0);
+        (,uint256 chop,, uint256 dirt) = dog.ilks(ilk);
+        assertEq(dirt, 0);
+
+        dog.bark(ilk, me);
+
+        (, uint256 tab, uint256 lot,,,) = clip.sales(1);
+
+        (, uint256 rate,,,) = vat.ilks(ilk);
+
+        assertEq(lot, 40 ether * (tab * 1 ether / rate / chop) / 100 ether);
+        assertEq(tab, rad(75 ether) - ray(0.2 ether)); // 0.2 RAY rounding error
+
+        assertEq(_ink(ilk, me), 40 ether - lot);
+        assertEq(_art(ilk, me), 100 ether - tab * WAD / rate / chop);
+
+        assertEq(dog.Dirt(), tab);
+        (,,, dirt) = dog.ilks(ilk);
+        assertEq(dirt, tab);
+    }
+
+    function test_partial_liquidation_hole_limit() public {
+        dog.file(ilk, "hole", rad(75 ether));
+
+        assertEq(_ink(ilk, me), 40 ether);
+        assertEq(_art(ilk, me), 100 ether);
+
+        assertEq(dog.Dirt(), 0);
+        (,uint256 chop,, uint256 dirt) = dog.ilks(ilk);
+        assertEq(dirt, 0);
+
+        dog.bark(ilk, me);
+
+        (, uint256 tab, uint256 lot,,,) = clip.sales(1);
+
+        (, uint256 rate,,,) = vat.ilks(ilk);
+
+        assertEq(lot, 40 ether * (tab * 1 ether / rate / chop) / 100 ether);
+        assertEq(tab, rad(75 ether) - 0.2 * 10 ** 27); // 2* 10 ** 26 rounding error
```

Comment:
Inconsistent with line 507--`ray(0.2 ether)` versus `0.2 * 10 ** 27`. I prefer either `ray(0.2 ether)` or `0.2 * RAY`.

Reply:
Fixed, had changed that after but forgot about the other case.